x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/relevance_tuning_form/relevance_tuning_item_content/weight_slider.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useActions } from 'kea';
+
+import { EuiFormRow, EuiRange } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { RelevanceTuningLogic } from '../../relevance_tuning_logic';
+import { SearchField } from '../../types';
+
+interface Props {
+  name: string;
+  field: SearchField;
+}
+
+export const WeightSlider: React.FC<Props> = ({ name, field }) => {
+  const { updateFieldWeight } = useActions(RelevanceTuningLogic);
+
+  return (
+    <EuiFormRow
+      label={i18n.translate(
+        'xpack.enterpriseSearch.appSearch.engine.relevanceTuning.manageFields.weight.label',
+        {
+          defaultMessage: 'Weight',
+        }
+      )}
+    >
+      <EuiRange
+        min={0}
+        max={10}
+        step={0.1}
+        value={field.weight}
+        onChange={(e) =>
+          updateFieldWeight(
+            name,
+            parseFloat((e as React.ChangeEvent<HTMLInputElement>).target.value)
+          )
+        }
```

Comment:
I believe you should be able to put the `e` typing in the arg vs `as`

```suggestion
        onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
          updateFieldWeight(name, parseFloat(e.target.value))
        }
```

Reply:
You would think that would work, but it doesn't. It complains that `.value` doesn't exist on the target object.

I wondered to myself... "how else do they deal with this in Kibana?"

https://github.com/elastic/kibana/blob/master/x-pack/plugins/maps/public/classes/styles/vector/components/data_mapping/ordinal_data_mapping_popover.tsx#L99

https://github.com/elastic/kibana/blob/master/x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/configuration_step/configuration_step_form.tsx#L517

https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/public/detections/components/rules/anomaly_threshold_slider/index.tsx#L27

https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/public/resolver/view/graph_controls.tsx#L142

It looks like everyone else is doing something similar.

Reply:
Ahh you're totally right, and super interesting to see other usages within Kibana have the same issue. I wonder if we should report it to the EUI folks.

Reply:
Could be a good idea

Reply:
I entered an issue, we'll see what they say: https://github.com/elastic/eui/issues/4533